great_expectations/checkpoint/checkpoint.py

```diff
@@ -127,8 +220,66 @@ def run(
             "validations": validations,
             "profilers": profilers,
         }
-        substituted_config = self.get_substituted_config(runtime_kwargs=runtime_kwargs)
-        # TODO: <Alex>Rob to complete</Alex>
+        substituted_runtime_config: CheckpointConfig = self.get_substituted_config(
+            runtime_kwargs=runtime_kwargs
+        )
+        run_name_template: Union[
+            str, None
+        ] = substituted_runtime_config.run_name_template
+        validations: list = substituted_runtime_config.validations
+        results = []
+
+        if run_name is None and run_name_template is not None:
+            run_name: str = self.get_run_name_from_template(
+                run_name_template=run_name_template
+            )
+
+        for idx, validation_dict in enumerate(validations):
+            try:
+                substituted_validation_dict: dict = (
+                    self.get_substituted_validation_dict(
+                        substituted_runtime_config=substituted_runtime_config,
+                        validation_dict=validation_dict,
+                    )
+                )
+                batch_request: BatchRequest = substituted_validation_dict.get(
+                    "batch_request"
+                )
+                expectation_suite_name: str = substituted_validation_dict.get(
+                    "expectation_suite_name"
+                )
+                action_list: list = substituted_validation_dict.get("action_list")
+
+                validator: Validator = self.data_context.get_validator(
+                    batch_request=batch_request,
+                    expectation_suite_name=expectation_suite_name,
+                )
+                action_list_validation_operator = ActionListValidationOperator(
```

Comment:
How about a typehint here -- we have been so diligent about them...

Reply:
Added.